From ad93d21aff4ea68297b0cc167e8490ba8c5fc8e1 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Fri, 14 Dec 2018 15:04:38 +0530
Subject: [PATCH] Revert "Fail some impossible cross-detections"

This partially reverts commit 26ff712bae08e8dfc76b3e424b0b5c2b61560158.

GStreamer relied on this broken behaviour to detect iOS frameworks.
---
 mesonbuild/dependencies/base.py |  5 -----
 mesonbuild/dependencies/misc.py | 10 +---------
 2 files changed, 1 insertion(+), 14 deletions(-)

diff --git a/mesonbuild/dependencies/base.py b/mesonbuild/dependencies/base.py
index cd029392..ab3f1556 100644
--- a/mesonbuild/dependencies/base.py
+++ b/mesonbuild/dependencies/base.py
@@ -1934,11 +1934,6 @@ class ExtraFrameworkDependency(ExternalDependency):
             self.link_args = ['-F' + self.path, '-framework', self.name.split('.')[0]]
 
     def detect(self, name, path):
-        # should use the compiler to look for frameworks, rather than peering at
-        # the filesystem, so we can also find them when cross-compiling
-        if self.want_cross:
-            return
-
         lname = name.lower()
         if path is None:
             paths = ['/System/Library/Frameworks', '/Library/Frameworks']
-- 
2.17.2 (Apple Git-113)

